lib/g3-types/src/net/openssl/client/mod.rs

```diff
@@ -371,7 +378,19 @@ impl OpensslClientConfigBuilder {
     fn new_tls13_builder(&self) -> anyhow::Result<SslConnectorBuilder> {
         let mut ctx_builder = SslConnector::builder(SslMethod::tls_client())
             .map_err(|e| anyhow!("failed to create ssl context builder: {e}"))?;
-        ctx_builder.set_verify(SslVerifyMode::PEER);
+
+        if self.insecure {
+            warn!("Tls Insecure Mode: Tls Peer (server) cert vertification is no longer enforced for this Context!");
+            ctx_builder.set_verify(SslVerifyMode::NONE);
+            ctx_builder.set_verify_callback(SslVerifyMode::NONE, |valid, _ctx_ref| {
+                if !valid {
+                    warn!("ssl verification failed but enforcment disabled");
+                }
+                true
+            });
+        } else {
+            ctx_builder.set_verify(SslVerifyMode::PEER);
+        }
```

Comment:
The can be moved as a new function `self.set_verify(&mut ctx_builder)`.
There will be 5 lines (310, 347, 374, 399, 425) in this file need to be updated.